SDWebImage/SDMemoryLRUCache.m

```diff
@@ -0,0 +1,495 @@
+/*
+ * This file is part of the SDWebImage package.
+ * (c) Olivier Poitrey <rs@dailymotion.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+
+#import "SDMemoryLRUCache.h"
+#import "SDImageCacheConfig.h"
+#import "SDInternalMacros.h"
+#import "UIImage+MemoryCacheCost.h"
+#import <CoreFoundation/CoreFoundation.h>
+#import <pthread.h>
+
+static void *SDMemoryLRUCacheContext = &SDMemoryLRUCacheContext;
+
+static inline dispatch_queue_t SDMemoryCacheGetReleaseQueue() {
```

Comment:
Any future plan or reason, to always using a random `global queue` to release the node object?

For example, can we create a concurrent queue (one per LRU cache) to release ? Or is this user-configable ?

Reply:
I saw [YYMemoryCache](https://github.com/ibireme/YYCache/blob/master/YYCache/YYMemoryCache.m)'s original code have this method. But they also have one `releaseOnMainThread` and `releaseAsynchronously`. We just remove this and use `releaseOnMainThread = NO` && `releaseAsynchronously`. This may not a always a good idea in practice. (I can provide you some tech detail for this, do we need to discuss on WeChat ?)